### PR TITLE
added wallhaven.cc

### PR DESCRIPTION
**Pull request by Venipa**
features:
- searching, top list, random & latest
image

- viewing wallpaper
image

- open wallpaper button (could not test it yet, only on canary discord?)
- reading forum/thread/board
image

### CODE REVIEW

File: websites/W/Wallhaven/presence.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        data.state = "Viewing...";
+        data.details =
+          tags.length > 0
+            ? tags.slice(0, 3).join(", ")
+            : "Wallpaper has no tags";
+
+        if (showButton) {
+          data.buttons = [
+            {
+              label: "Open Wallpaper",
```
Comment (ririxidev) [inline websites/W/Wallhaven/presence.ts:126]: Open Wallpaper URL

File: websites/W/Wallhaven/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "general",
+    "wallpaper",
+    "wallhaven",
+    "wh",
+    "whcc"
+  ],
+  "settings": [
+    {
+      "id": "buttons",
+      "title": "Share Wallpaper URL",
```
Comment (ririxidev) [inline websites/W/Wallhaven/dist/metadata.json:28]: Show button for wallpaper url

File: websites/W/Wallhaven/presence.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      } as PresenceData;
+      if (document.location.hostname == "wallhaven.cc") {
+        const query: { [key: string]: any } = getQuery();
+        const strings: { [key: string]: string } = await app.getStrings({
+            play: "presence.playback.playing",
+            pause: "presence.playback.paused",
+            browsing: "presence.activity.browsing",
+            searching: "presence.activity.searching",
+            reading: "presence.activity.reading",
+            idling: "Idling..."
```
Comment (ririxidev) [inline websites/W/Wallhaven/presence.ts:165]: it isn't valid translation string...

**Review by ririxidev — CHANGES_REQUESTED** [review]
you should look at this:

**Comment by Venipa** [comment]
@ririxidev i've applied the requested changes and applied codacy's suggestion

**Review by ririxidev — DISMISSED** [review]
looks fine for me 🤷

File: websites/W/Wallhaven/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  "description": {
+    "en": "Your source for the best high quality wallpapers on the Net!"
+  },
+  "url": "wallhaven.cc",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/SsIZl80.png",
+  "thumbnail": "https://i.imgur.com/QV6vAk6.jpg",
+  "color": "#0E6288",
+  "iframe": false,
+  "tags": [
```
Comment (Bas950) [inline websites/W/Wallhaven/dist/metadata.json:17]: no need to set it to false since thats the default value

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/W/Wallhaven/dist/metadata.json (hunk 4 above)
Comment (Venipa) [reply websites/W/Wallhaven/dist/metadata.json]: I've committed the requested changes 👌

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**